pkg/doc/bbs/bbs12381g2pub/internal/kilic/bls12-381/g1_custom.go

```diff
@@ -0,0 +1,512 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package bls12381
+
+import (
+	"hash"
+)
+
+func HashToCurve(msg, domain []byte, hashFunc func() hash.Hash) ([]byte, error) {
+	hashRes, err := hashToFpXMD(hashFunc, msg, domain, 2)
+	if err != nil {
+		return nil, err
+	}
+	u0, u1 := hashRes[0], hashRes[1]
+
+	p0, p1 := osswuMap(u0), osswuMap(u1)
+	Add(p0, p0, p1)
+	Affine(p0)
+	isogenyMapG1(&p0[0], &p0[1])
+	ClearCofactor(p0)
+	return ToBytes(Affine(p0)), nil
+}
+
+// Algorithm is re-written from Rust code (https://github.com/algorand/pairing-plus/blob/7ec2ae03aae4ba2fc5210810211478171ccededf/src/bls12_381/osswu_map/g1.rs#L48).
```

Comment:
Please note the issue number at kilic.

Do you have a summary of the differences in the code and outputs?

Reply:
I have not created the issue yet as I am not sure about at what side (kilic vs libursa) the error is present. I will create it later.